### PR TITLE
Charts: Adding a theme provider

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Fixes https://github.com/Automattic/jetpack-roadmap/issues/2041

## Proposed changes:

* extending Automattic Charts with a theme provider, allowing for easier chart adoption
* replacing hard-coded colours in the chart components

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* install Jetpack packages if needed (this PR doesn't add new packages) `pnpm i`
* navigate to the directory and run storybook: `cd projects/js-packages/charts` and `pnpm run storybook`
* verify that existing charts are working and new theme provider is available
* verify that different themes load different colours

SCR-20241211-mwct

### CODE REVIEW

File: projects/js-packages/charts/src/providers/theme/themes.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	gridColor: '',
+	gridColorDark: '',
+};
+
+/**
+ * Jetpack theme configuration
+ */
+const jetpackTheme: ChartTheme = {
+	backgroundColor: '#FFFFFF',
+	colors: [ '#98C8DF', '#006DAB', '#A6DC80', '#1F9828', '#FF8C8F' ],
```
Comment (annacmc) [inline projects/js-packages/charts/src/providers/theme/themes.ts:23]: Can we import color vars from `color-studio` for these themes, or would that be adding unnecessary extra dependencies?

**Review by annacmc — DISMISSED** [review]
This is awesome, I love it. Added one small, non-blocking, thought/question but otherwise it all works as expected and is very cool. Thank you for working on this 🙌

**Review by CodeyGuyDylan — DISMISSED** [review]
This looks great to me!

File: projects/js-packages/charts/src/providers/theme/themes.ts (hunk 1 above)
Comment (grzegorz-cp) [reply projects/js-packages/charts/src/providers/theme/themes.ts:23]: Good point. I'll add a task to check this and see how/if we can leverage CSS variables for it.

**Pull request merged**